src/utils/regex.cc

```diff
@@ -115,6 +133,66 @@ bool Regex::searchOneMatch(const std::string& s, std::vector<SMatchCapture>& cap
     return (rc > 0);
 }
 
+bool Regex::searchGlobal(const std::string& s, std::vector<SMatchCapture>& captures) const {
+    const char *subject = s.c_str();
+
+    bool prev_match_zero_length = false;
+    int pcre_options = 0;
```

Comment:
int pcre_options scope could be reduce. Having it inside the while loop - 

https://github.com/SpiderLabs/ModSecurity/blob/2672db103e511cbf0866efc62dc7a3b4f43a8657/src/utils/regex.cc#L139-L147